### ISSUE TITLE
[CRASH] --enable-aesni in WolfSSL, will crash opensips.

### ISSUE THREAD

**Issue by WangHoi** [OpenSIPS/opensips#2546]
OS: centos7 x86_64 with cpu(intel-xeon) support AES-NI
Version: opensips 3.2 branch, compiled wolfssl with --enable-aesni --enable-intelasm
Reason: F_MALLOC, allocated memory not 16 bytes aligned.

Inside aes_asm.S:

```
/*
void AES_256_Key_Expansion (const unsigned char *userkey,
  unsigned char *key)
*/

# parameter 2: %rsi

movdqa %xmm1, (%rsi)
```

$rsi not aligned.

**Comment by liviuchircu** [comment]
Hi, @WangHoi!  I pushed a fix on OpenSIPS `3.1` and `3.2` that will allow you to customize the alignment and recompile, if you want to keep using `F_MALLOC` (change [this macro](https://github.com/OpenSIPS/opensips/blob/master/mem/f_malloc.h#L40) to `16UL`).

However, I would really recommend simply using the `-a Q_MALLOC` allocator.  Just pass this command-line option at startup, and you should be good.  It is natively 16-byte aligned, so you should not run into this trouble anymore!

Cheers,

**Comment by liviuchircu** [comment]
Important update here: per commit c29652fa10, we decided that the default alignment of `Q_MALLOC` should remain `8` bytes, to be compatible with all other allocators.  So in order to have 16-byte aligned memory, as required by your custom build of WolfSSL, you cannot use the default OpenSIPS packages and must use a custom build, where you change the OpenSIPS `F_MALLOC` allocator to be 16-byte aligned, by changing [this line](https://github.com/OpenSIPS/opensips/blob/master/mem/f_malloc.h#L40).

If you run into any more trouble on the _alignment_ topic, we can continue the discussion here, no need to open a new issue.  Cheers!
